packages/evm-connector/src/EvmWalletConnector.ts

```diff
@@ -0,0 +1,461 @@
+import { hexToBytes } from '@ethereumjs/util';
+// External libraries
+import {
+  type BytesLike,
+  arrayify,
+  hexlify,
+  splitSignature,
+} from '@ethersproject/bytes';
+import memoize from 'memoizee';
+
+import {
+  type AbiMap,
+  Address,
+  type Asset,
+  type ConnectorMetadata,
+  FuelConnector,
+  FuelConnectorEventTypes,
+  type InputValue,
+  type JsonAbi,
+  type Network,
+  Predicate,
+  Provider,
+  type TransactionRequestLike,
+  type Version,
+  getPredicateRoot,
+  transactionRequestify,
+} from 'fuels';
+
+import { BETA_5_URL, WINDOW } from './constants';
+import {
+  type EVMWalletConnectorConfig,
+  EVMWalletConnectorEvents,
+} from './types';
+import type { EIP1193Provider } from './utils/eip-1193';
+import { METAMASK_ICON } from './utils/metamask-icon';
+import { predicates } from './utils/predicateResources';
+
+export class EVMWalletConnector extends FuelConnector {
+  name = 'EVM wallet connector';
+  metadata: ConnectorMetadata = {
+    image: METAMASK_ICON,
+    install: {
+      action: 'Install',
+      description: 'Install a ethereum Wallet to connect to Fuel',
+      link: 'https://ethereum.org/en/wallets/find-wallet/',
+    },
+  };
+
+  installed = true;
+  connected = false;
+
+  ethProvider: EIP1193Provider | null = null;
+  fuelProvider: Provider | null = null;
+
+  events = {
+    ...FuelConnectorEventTypes,
+    ...EVMWalletConnectorEvents,
+  };
+
+  private predicate = predicates['verification-predicate'];
+  private setupLock = false;
+  private _currentAccount: string | null = null;
+  private config: Required<EVMWalletConnectorConfig>;
+  private _ethereumEvents = 0;
+
+  constructor(config: EVMWalletConnectorConfig = {}) {
+    super();
+
+    this.config = Object.assign(config, {
+      fuelProvider: config.fuelProvider || BETA_5_URL,
```

Comment:
I think we should turn config.fuelProvider to only accept a fuel Provider instance this would also clean up a bit the logic on line 115 to 118.